src/lib/server/auth.ts

```diff
@@ -0,0 +1,69 @@
+import { SvelteKitAuth } from '@auth/sveltekit';
+import { PrismaAdapter } from '@auth/prisma-adapter';
+
+import { env } from '$env/dynamic/private';
+import type { OIDCConfig } from '@auth/sveltekit/providers';
+import prisma from './db/prisma';
+import { error } from '@sveltejs/kit';
+
+interface SurfConextProfile extends Record<string, any> {
+	nickname: string;
+	firstName: string;
+	lastName: string;
+	email: string;
+}
+
+/**
+ * Workaround for Auth.js not automatically fetching userinfo through endpoint specified in .well-known config
+ *
+ * @param accessToken JWT access token
+ * @returns Object containing user information
+ */
+async function fetchUserInfo(accessToken: string | undefined) {
+	if (!accessToken) throw error(505, 'No access token provided');
+	const res = await fetch(`${env.SURFCONEXT_ISSUER}/oidc/userinfo`, {
+		headers: {
+			Authorization: `Bearer ${accessToken}`
+		}
+	});
+	return await res.json();
+}
+
+function SurfConextProvider<P extends SurfConextProfile>(): OIDCConfig<P> {
+	return {
+		id: 'surfconext',
+		name: 'SURFconext',
+		type: 'oidc',
+		issuer: env.SURFCONEXT_ISSUER,
+		wellKnown: `${env.SURFCONEXT_ISSUER}/.well-known/openid-configuration`,
+		clientId: env.SURFCONEXT_CLIENT_ID,
+		clientSecret: env.SURFCONEXT_CLIENT_SECRET,
+		allowDangerousEmailAccountLinking: true, // Not sure if this is safe @juliavdkris
```

Comment:
@juliavdkris

Reply:
It's a workaround for a problem I was having where it tried to create a new user every time you logged in. If you trust the authentication providers to verify emails, it's fine to use. And this is the case for us, since there's only one authentication provider and it's the TU Delft.

See the docs: https://authjs.dev/reference/core/providers#allowdangerousemailaccountlinking

Reply:
Nice 😄